### PR TITLE
Add more content about granting CloudDNS permission to GKE cluster

### PR DESCRIPTION
**Pull request by ZhiminXiang**
Fixes #(issue-number)

## Proposed Changes
* rename using-external-dns to using-external-dns-on-gcp since this doc is GCP specific.
* Add the content of granting cluster the CloudDNS permission by using service account credential
* Directly paste the manifest of ExternalDNS into the doc instead of asking users to find out from ExternalDNS site.

### CODE REVIEW

File: serving/using-external-dns-on-gcp.md [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+```shell
+gcloud dns record-sets transaction start --zone "my-org-do"
+gcloud dns record-sets transaction add ns-cloud-e{1..4}.googledomains.com. \
+    --name "external-dns-test.my-org.do." --ttl 300 --type NS --zone "my-org-do"
+gcloud dns record-sets transaction execute --zone "my-org-do"
+```
+
+### Deploy ExternalDNS
+
+1. Use below manifest if you set up your cluster with CloudDNS scope (by following [the instruction](#cluster-with-cloud-dns-scope)).
```
[the remark this answers is not in the document]
Comment (ZhiminXiang) [reply serving/using-external-dns-on-gcp.md]: Done.

File: serving/using-external-dns-on-gcp.md [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+steps to add the NS records of this zone into the parent zone: 
+```shell
+gcloud dns record-sets transaction start --zone "my-org-do"
+gcloud dns record-sets transaction add ns-cloud-e{1..4}.googledomains.com. \
+    --name "external-dns-test.my-org.do." --ttl 300 --type NS --zone "my-org-do"
+gcloud dns record-sets transaction execute --zone "my-org-do"
+```
+
+### Deploy ExternalDNS
+
```
[the remark this answers is not in the document]
Comment (ZhiminXiang) [reply serving/using-external-dns-on-gcp.md]: Done.

**Comment by ZhiminXiang** [comment]
friendly ping @averikitsch :)

File: serving/using-external-dns-on-gcp.md [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+    --machine-type=n1-standard-4 \
+    --enable-autoscaling --min-nodes=1 --max-nodes=10 \
+    --enable-autorepair \
+    --scopes=service-control,service-management,compute-rw,storage-ro,cloud-platform,logging-write,monitoring-write,pubsub,datastore \
+    --num-nodes=3
+```
+
+2. Create a new service account with the project role `dns.admin`:
+```shell
+# Name of the service account you want to create.
```
Comment (averikitsch) [inline serving/using-external-dns-on-gcp.md]: To be more specific, please make each comment a step.

File: serving/using-external-dns-on-gcp.md [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+    --name "external-dns-test.my-org.do." --ttl 300 --type NS --zone "my-org-do"
+gcloud dns record-sets transaction execute --zone "my-org-do"
+```
+
+### Deploy ExternalDNS
+
+Firstly, choose the manifest of ExternalDNS.
+
+Use below manifest if you set up your cluster with [CloudDNS scope](#cluster-with-cloud-dns-scope).
+```
```
Comment (averikitsch) [inline serving/using-external-dns-on-gcp.md]: add yaml here for correct rendering and below.

File: serving/using-external-dns-on-gcp.md (hunk 3 above)
Comment (ZhiminXiang) [reply serving/using-external-dns-on-gcp.md]: Done

File: serving/using-external-dns-on-gcp.md (hunk 4 above)
Comment (ZhiminXiang) [reply serving/using-external-dns-on-gcp.md]: Done

**Comment by averikitsch** [comment]
@ZhiminXiang please fix the merge conflict

**Pull request merged**